comp/core/gui/guiimpl/gui.go

```diff
@@ -86,66 +83,74 @@ type dependencies struct {
 	Lc        fx.Lifecycle
 }
 
+type provides struct {
+	fx.Out
+
+	Comp     optional.Option[guicomp.Component]
+	Endpoint api.AgentEndpointProvider
+}
+
 // GUI component implementation constructor
 // @param deps dependencies needed to construct the gui, bundled in a struct
 // @return an optional, depending of "GUI_port" configuration value
-func newGui(deps dependencies) optional.Option[guicomp.Component] {
+func newGui(deps dependencies) provides {
 
+	p := provides{
+		Comp:     optional.NewNoneOption[guicomp.Component](),
+		Endpoint: api.NewAgentEndpointProvider(func(_ http.ResponseWriter, _ *http.Request) {}, "/gui/intent", "GET"),
```

Comment:
In the case of `guiPort` being `-1` we are still registering the `/gui/intent` endpoint. I think it would make more sense to not register an endpoint in the case of the GUI being disable.

I just open an PR that allow us to pass nil endpoints 😄

Reply:
That's much nicer that way, thanks ! Done in 269022d16e493c88bdcdb13579f4cf583e2e6031.